apps/emqx_dashboard/src/emqx_dashboard_token.erl

```diff
@@ -254,15 +247,31 @@ clean_expired_jwt(Now) ->
     ok = destroy(JWTList).
 
 -if(?EMQX_RELEASE_EDITION == ee).
-check_rbac(Req, Extra) ->
-    emqx_dashboard_rbac:check_rbac(Req, Extra).
+check_rbac(Req, JWT) ->
+    #?ADMIN_JWT{exptime = _ExpTime, extra = Extra, username = Username} = JWT,
+    case emqx_dashboard_rbac:check_rbac(Req, Extra) of
+        true ->
+            NewJWT = JWT#?ADMIN_JWT{exptime = jwt_expiration_time()},
+            {atomic, Res} = mria:transaction(
+                ?DASHBOARD_SHARD,
+                fun mnesia:write/1,
+                [NewJWT]
+            ),
+            {Res, Username};
+        _ ->
+            {error, unauthorized_role}
+    end.
 
 -else.
 
--dialyzer({nowarn_function, [check_rbac/2]}).
--dialyzer({no_match, [do_verify/2]}).
-
-check_rbac(_Req, _Extra) ->
-    true.
+check_rbac(_Req, JWT) ->
+    #?ADMIN_JWT{exptime = _ExpTime, extra = _Extra, username = Username} = JWT,
```

Comment:
Why not extract this as a shared function

Reply:
:heavy_check_mark: